src/main.c

```diff
@@ -4,13 +4,20 @@
 
 #include "password.h"
 
+#define DEFAULT_LENGTH 4
+
 int main(int argc, char **argv)
 {
   srand(time(0));
-  int len = 4;
-  if(argc > 1)
+  int len;
+  if(argc > 1) {
     len = atoi(argv[1]);
-
+    if (len < 0) {
+      len = DEFAULT_LENGTH;
+    }
+  } else {
+    len = DEFAULT_LENGTH;
+  }
```

Comment:
Sorry for being a year late. I was looking through my old projects and randomly came across this pull request. I seem to have never got a notification for it. 

Anyways, this is a bit overcomplicated. For example, there's no reason to have the else block at the end--you could just set `len` to its default value at the beginning and only change it if it's warranted. That way, you can remove the else block at the end and adhere to the DRY principle.